src/freenet/clients/http/SecurityLevelsToadlet.java

```diff
@@ -144,7 +146,9 @@ public void handleMethodPOST(URI uri, HTTPRequest request, ToadletContext ctx) t
 					if(newPhysicalLevel == PHYSICAL_THREAT_LEVEL.HIGH && node.securityLevels.getPhysicalThreatLevel() != newPhysicalLevel) {
 						// Check for password
 						String password = request.getPartAsStringFailsafe("masterPassword", MAX_PASSWORD_LENGTH);
-						if(password != null && password.length() > 0) {
+						String confirmPassword = request.getPartAsStringFailsafe("confirmMasterPassword", MAX_PASSWORD_LENGTH);
+						boolean passwordsMatch = password.equals(confirmPassword);
+						if(password != null && confirmPassword != null && password.length() > 0 && confirmPassword.length() > 0 && passwordsMatch) {
```

Comment:
`!password.isEmpty()` seems more semantic than a length check; same sort of thing for the `password.length() == 0`.